pkg/scale/bitvec.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package scale
+
+const (
+	// maxLen equivalent of `ARCH32BIT_BITSLICE_MAX_BITS` in parity-scale-codec
+	maxLen = 268435455
+	// bitSize is the number of bits in a byte
+	bitSize = 8
+)
+
+// BitVec represents rust's `bitvec::BitVec` in SCALE
+// It is encoded as a compact u32 representing the number of bits in the vector
+// followed by the actual bits, rounded up to the nearest byte
+type BitVec interface {
+	// Bits returns the bits in the BitVec
+	Bits() []uint8
+	// Bytes returns the byte representation of the Bits
+	Bytes() []byte
+	// Size returns the number of bits in the BitVec
+	Size() uint
+}
+
+// bitVec implements BitVec
+type bitVec struct {
+	size uint    `scale:"1"`
+	bits []uint8 `scale:"2"`
+}
+
+// NewBitVec returns a new BitVec with the given bits
+func NewBitVec(bits []uint8) BitVec {
+	var size uint
+	if bits != nil {
+		size = uint(len(bits))
+	}
+
+	return &bitVec{
+		size: size,
+		bits: bits,
+	}
+}
+
+// Bits returns the bits in the BitVec
+func (bv *bitVec) Bits() []uint8 {
+	return bv.bits
+}
+
+// Bytes returns the byte representation of the BitVec.Bits
+func (bv *bitVec) Bytes() []byte {
+	var b []byte
+	for i := uint(0); i < bv.size; i += bitSize {
+		end := i + bitSize
+		if end > bv.size {
+			end = bv.size
+		}
+		chunk := bv.bits[i:end]
+		b = append(b, bitsToBytes(chunk)...)
+	}
+	return b
```

Comment:
I would say you don't need to create chunks since the function `bitsToBytes` already do that, so you can pass the entire bit slice to it
```suggestion
	return bitsToBytes(bv.bits)
```